autoreduce_qp/queue_processor/reduction/process_manager.py

```diff
@@ -27,33 +26,88 @@ def __init__(self, message: Message, run_name: str) -> None:
         try:
-            # We need to run the reduction in a new process, otherwise scripts
-            # will fail when they use things that require access to a main loop
-            # e.g. a GUI main loop, for matplotlib or Mantid
-            python_path = sys.executable
-            with tempfile.NamedTemporaryFile("w+") as temp_output_file:
-                args = [python_path, RUNNER_PATH, self.message.serialize(), temp_output_file.name, self.run_name]
-                logger.info("Calling: %s %s %s %s %s", python_path, RUNNER_PATH,
-                            self.message.serialize(limit_reduction_script=True), temp_output_file.name, self.run_name)
-
-                # Copy and update the subprocess environment to inherit the
-                # parent one, and append the PYTHONPATH of the queue_processor module
-                environment = os.environ.copy()
-
-                pythonpath = environment.get("PYTHONPATH", "")
-                pythonpath += f':{RUNNER_PATH.split("autoreduce_qp")[0]}'
-                pythonpath = environment["PYTHONPATH"] = pythonpath
-
-                # Run process until finished and check the exit code for success
-                subprocess.run(args, check=True, env=environment)
-
-                # The subprocess will write out the result message in the
-                # tempfile, read it back
-                result_message_raw = temp_output_file.file.read()
-
-            result_message = Message()
-            result_message.populate(result_message_raw)
-        except subprocess.CalledProcessError:
+            # Create a temp directory to store reduction result
+            # Volume bind for the container to write to and host to read from
+            with tempfile.TemporaryDirectory() as temp_dir:
+                os.chmod(temp_dir, 0o777)
+
+                # We need to run the reduction in a new process, otherwise scripts
+                # will fail when they use things that require access to a main loop
+                # e.g. a GUI main loop, for matplotlib or Mantid
+                serialized_vars = self.message.serialize()
+                serialized_vars_truncated = self.message.serialize(limit_reduction_script=True)
+                args = ["python3", "runner.py", serialized_vars, self.run_name]
+                logger.info("Calling: %s %s %s %s ", "python3", "runner.py", serialized_vars_truncated, self.run_name)
+
+                # Return a client configured from environment variables
+                # The environment variables used are the same as those used by the Docker command-line client
+                # https://docs.docker.com/engine/reference/commandline/cli/#environment-variables
+                client = docker.from_env()
+
+                # Pull all tags of runner-mantid as a list
+                # Could be used to populate a dropdown menu of images
+                images = client.images.pull('ghcr.io/autoreduction/runner-mantid', all_tags=True)
```

Comment:
The issue I see with `all_tags=True` is that if you push a development one then it'll be pulled onto the production qp machines. While not terrible, it seems like an unnecessary waste of disk space.

And ideally software will be selected either in reduce_vars (with a new special attribute), or from the webapp (which will fill the options from the DB, rather than communicating with the queue processor)